test/functional/test_framework/script.py

```diff
@@ -849,7 +849,7 @@ def TaprootSignatureMsg(txTo, spent_utxos, hash_type, input_index=0, *, scriptpa
     if scriptpath:
         ss += TaggedHash("TapLeaf", bytes([leaf_ver]) + ser_string(leaf_script))
         ss += bytes([0])
-        ss += codeseparator_pos.to_bytes(4, "little", signed=True)
+        ss += codeseparator_pos.to_bytes(4, "little", signed=False)
```

Comment:
Why this change? Having a code separator at position `2**31` or more doesn't seem very plausible given the block weight limit is lower than `2**22`, and writing `0xffff_ffff` as -1 and `0xffff_fffe` as -2 seems fine?

Reply:
> Having a code separator at position 2**31 or more doesn't seem very plausible given the block weight limit is lower than 2**22

This might be an argument for more coverage, but not against adding coverage. Obviously the case will never make it into the "success" column without a a hardfork.

> writing 0xffff_ffff as -1 and 0xffff_fffe as -2 seems fine?

I think writing test code to be obvious is better, all else equal.